src/plugins/showMeYourName/index.tsx

```diff
@@ -4,48 +4,236 @@
+        default: "",
+        isValid: validColor,
+    },
+    alwaysShowDisplaySymbols: {
         type: OptionType.BOOLEAN,
-        description: "Use display names in place of usernames",
-        default: false
+        description: "Show display name prefix and suffix even if the display name is the only name displayed, or is the first displayed.",
+        default: false,
+    },
+    alwaysShowDisplayColor: {
+        type: OptionType.BOOLEAN,
+        description: "Show display name color even if the display name is the only name displayed, or is the first displayed.",
+        default: false,
+    },
+    usernamePrefix: {
+        type: OptionType.STRING,
+        description: "The symbol to use as a prefix for the username. Can be up to 3 non-alphanumeric characters long.",
+        default: "(@",
+        isValid: validSymbols,
     },
-    inReplies: {
+    usernameSuffix: {
+        type: OptionType.STRING,
+        description: "The symbol to use as a suffix for the username. Can be up to 3 non-alphanumeric characters long.",
+        default: ")",
+        isValid: validSymbols,
+    },
+    usernameColor: {
+        type: OptionType.STRING,
+        description: "The color to use for the username. Accepts hex(a), rgb(a), or hsl(a) input. Leave blank for default. \"Role\" to follow the user's top role color. \"Role+-#\" to adjust the brightness by that percentage (ex: \"Role+15\")",
+        default: "Role-25",
+        isValid: validColor,
+    },
+    usernameSymbolColor: {
+        type: OptionType.STRING,
+        description: "The color to use for username symbols. Accepts hex(a), rgb(a), or hsl(a) input. Leave blank for default. \"Role\" to follow the user's top role color. \"Role+-#\" to adjust the brightness by that percentage (ex: \"Role+15\")",
+        default: "",
+        isValid: validColor,
+    },
+    alwaysShowUsernameSymbols: {
         type: OptionType.BOOLEAN,
+        description: "Show username prefix and suffix even if the username is the only name displayed, or is the first displayed.",
         default: false,
-        description: "Also apply functionality to reply previews",
     },
+    alwaysShowUsernameColor: {
+        type: OptionType.BOOLEAN,
+        description: "Show username color even if the username is the only name displayed, or is the first displayed.",
+        default: false,
+    }
 });
 
 export default definePlugin({
     name: "ShowMeYourName",
-    description: "Display usernames next to nicks, or no nicks at all",
-    authors: [Devs.Rini, Devs.TheKodeToad],
+    description: "Display any permutation of nicknames, display names, and usernames in chat. Nicknames are per-server and display names are global. You will need to hover over old messages to update them after changing any settings.",
```

Comment:
```suggestion
    description: "Display any permutation of nicknames, display names, and usernames in chat",
```

describing nicknames and display names doesn't belong in the plugin description

if you don't want to hover over messages to update them after changing settings, you can use settings as react state